app/src/routes/v1/keycloak.js

```diff
@@ -0,0 +1,54 @@
+const config = require('config');
+
+const Problem = require('api-problem');
+
+const keycloak = require('../../components/keycloak');
+const KeyCloakServiceClientManager = require('../../components/keyCloakServiceClientMgr');
+const RealmAdminService = require('../../components/realmAdminSvc');
+
+const keycloakRouter = require('express').Router();
+
+// fetches all the service clients for all KC realms
+// current user must have role GETOK_ADMIN
+keycloakRouter.get('/serviceClients', keycloak.protect('realm:GETOK_ADMIN'), async (req, res) => {
+
+  const [devClients, testClients, prodClients] = await Promise.all([
+    getClientsFromEnv('dev'),
+    getClientsFromEnv('test'),
+    getClientsFromEnv('prod')
+  ]);
+
+  // join them all into one array
+  var result = devClients.concat(testClients, prodClients);
```

Comment:
Since we don't actually care about the discrete difference between dev, test and prod clients, you can do the same thing more succinctly:
```
const result = await Promise.all([
    getClientsFromEnv('dev'),
    getClientsFromEnv('test'),
    getClientsFromEnv('prod')
]).flat();
```
Also - use const or let. Do not use var for variable declarations.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat

Reply:
ok. not sure if you can flatten a promise. how about:
```
  const clients = await Promise.all([
    getClientsFromEnv('dev'),
    getClientsFromEnv('test'),
    getClientsFromEnv('prod')
  ]);
  // join them all into one array
  const result = clients.flat();
```